FrEIA/modules/base.py

```diff
@@ -0,0 +1,38 @@
+from typing import Tuple, Iterable, Union
+
+import torch.nn as nn
+from torch import Tensor, is_tensor
+
+
+class InvertibleOperator(nn.Module):
+    def __init__(self, dims_in: Tuple[int], dims_c: Tuple[int] = None):
+        super().__init__()
+        if dims_c is None:
+            dims_c = []
+        self.dims_in = dims_in
+        self.dims_c = dims_c
+
+    def forward(self, x: Union[Iterable[Tensor], Tensor], c: Union[Iterable[Tensor], Tensor], rev=False, jac=True):
+        # Tupelize input
+        is_tensor_input = is_tensor(x)
+        if is_tensor_input:
+            x = (x,)
+        if not is_tensor(c):
+            c = (c,)
+
+        # Compute
+        z, jac = self.compute(x, c, rev=rev, jac=jac)
+
+        # Check returned types -- todo unit test?
+        assert isinstance(z, tuple), f"{self.__class__.__name__}.compute(...) returned non-tuple vector"
```

Comment:
Alternative would be a unit test, but I guess that users subclassing this class would benefit from direct feedback.

Reply:
makes sense from my side